dtls.c

```diff
@@ -714,10 +715,30 @@ static const char *dtls_handshake_type_to_name(int type)
   case DTLS_HT_FINISHED:
     return "finished";
   default:
-    return "unknown";
+    snprintf(buf, sizeof(buf), "unknown (%d)", type);
```

Comment:
I would like to avoid `snprintf()` in tinydtls as it is not available on many non-POSIX embedded platforms. As this is used only for converting the handshake type number to ASCII characters, we could use a small helper function instead, e.g.:

```c
static size_t print_byte(char *buf, size_t buflen, uint8_t value) {
  char const *bufstart = buf;
  uint8_t quot = 100;

  if (value == 0) {
    *buf++ = '0';
    buflen--;
  } else {
    /* adjust quot to avoid leading zeroes */
    while (quot && (value / quot == 0))
      quot /= 10;

    while (quot && buflen) {
      *buf++ = '0' + (value / quot);
      buflen--;
      value %= quot;
      quot /= 10;
    }
  }
  return buf - bufstart;
}
```

Usage would be:

```c
  static char buf[16] = "unknown (";
  size_t pos = 9;

  pos += print_byte(buf + pos, sizeof(buf) - pos - 2, type & 0xff);
  assert(pos+2 < sizeof(buf));
  buf[pos++] = ')';
  buf[pos] = '\000';
```

Reply:
A hexdec value may do it as well ... and may be easier to `print`.

Reply:
Taken code suggestion.

Reply:
Just seen this

> A hexdec value may do it as well ... and may be easier to print.

Not familiar with hexdec, but I think I understand the principle.  `print_byte()` could be updated I guess.

Reply:
I thought about printing hex values but went for decimal as the handshake types in the RFC are specified in decimal.